framework/src/controller/jsonrpc/utils.ts

```diff
@@ -13,14 +13,21 @@
  */
 
 import { validator, LiskValidationError } from '@liskhq/lisk-validator';
-import { ErrorObject, JsonRpcError, ID, NotificationObject, Result, SuccessObject } from './types';
+import {
+	JsonRpcError,
+	ID,
+	NotificationRequest,
+	JsonRpcResult,
+	ResponseObjectWithResult,
+	ResponseObjectWithError,
+} from './types';
 
 export const VERSION = '2.0';
 
 const RequestSchema = {
 	id: 'jsonRPCRequestSchema',
 	type: 'object',
-	required: ['jsonrpc', 'method', 'id'],
```

Comment:
i think for a valid jsonrpc request the property id is required.. but the value can be string, number, null?
```
id
An identifier established by the Client that **MUST contain a String, Number, or NULL** value if included. If it is not included it is assumed to be a notification. The value SHOULD normally not be Null [1] and Numbers SHOULD NOT contain fractional parts [2]
```

Reply:
There is difference between request and notification. In next PR both had been segregated into separate schemas and this point is covered.